tests/test_schema.py

```diff
@@ -2348,51 +2349,67 @@ class ValidatingSchema(Schema):
-    is_collab = fields.Function(lambda user, ctx: user in ctx["blog"])
+    is_collab = fields.Function(
+        lambda user: user in typing.cast(dict, CONTEXT.get())["blog"]
+    )
 
     def get_is_owner(self, user):
-        return self.context["blog"].user.name == user.name
+        return CONTEXT.get()["blog"].user.name == user.name
 
 
 class TestContext:
+    def test_context_load_dump(self):
+        class ContextField(fields.Integer):
+            def _serialize(self, value, attr, obj, **kwargs):
+                value *= CONTEXT.get({}).get("factor", 1)
+                return super()._serialize(value, attr, obj, **kwargs)
+
+            def _deserialize(self, value, attr, data, **kwargs):
+                val = super()._deserialize(value, attr, data, **kwargs)
+                return val * CONTEXT.get({}).get("factor", 1)
+
+        class ContextSchema(Schema):
+            ctx_fld = ContextField()
+
+        ctx_schema = ContextSchema()
+
+        assert ctx_schema.load({"ctx_fld": 1}) == {"ctx_fld": 1}
+        assert ctx_schema.load({"ctx_fld": 1}, context={"factor": 2}) == {"ctx_fld": 2}
+        assert ctx_schema.dump({"ctx_fld": 1}) == {"ctx_fld": 1}
+        assert ctx_schema.dump({"ctx_fld": 1}, context={"factor": 2}) == {"ctx_fld": 2}
+        with Context({"factor": 3}):
+            assert ctx_schema.load({"ctx_fld": 1}) == {"ctx_fld": 3}
+            assert ctx_schema.load({"ctx_fld": 1}, context={"factor": 4}) == {
+                "ctx_fld": 4
+            }
+            assert ctx_schema.dump({"ctx_fld": 1}) == {"ctx_fld": 3}
+            assert ctx_schema.dump({"ctx_fld": 1}, context={"factor": 4}) == {
+                "ctx_fld": 4
+            }
+
     def test_context_method(self):
         owner = User("Joe")
         blog = Blog(title="Joe Blog", user=owner)
-        context = {"blog": blog}
         serializer = UserContextSchema()
-        serializer.context = context
-        data = serializer.dump(owner)
-        assert data["is_owner"] is True
-        nonowner = User("Fred")
-        data = serializer.dump(nonowner)
-        assert data["is_owner"] is False
+        with Context({"blog": blog}):
+            data = serializer.dump(owner)
+            assert data["is_owner"] is True
+            nonowner = User("Fred")
+            data = serializer.dump(nonowner)
+            assert data["is_owner"] is False
 
     def test_context_method_function(self):
         owner = User("Fred")
         blog = Blog("Killer Queen", user=owner)
         collab = User("Brian")
         blog.collaborators.append(collab)
-        context = {"blog": blog}
-        serializer = UserContextSchema()
-        serializer.context = context
-        data = serializer.dump(collab)
-        assert data["is_collab"] is True
-        noncollab = User("Foo")
-        data = serializer.dump(noncollab)
-        assert data["is_collab"] is False
-
-    def test_function_field_raises_error_when_context_not_available(self):
```

Comment:
This does not make much sense anymore. User code should manage the case where the context is unset.